### PR TITLE
Facebook and Twitter social icons in the share dialog

### PR DESCRIPTION
**Pull request by microsoftsam**
Simple Facebook and Twitter social icons in the share dialog that open up new popups into the Facebook sharer / Twitter inject pages: 

socialicons

Twitter message: "Check out what I made with @twitterHandle and @orgTwitterHandle! LINK"

### CODE REVIEW

**Comment by microsoftsam** [comment]
@pelikhan we really need to work on our fb and twitter meta tags, so it doesn't show the consent message as the description on no icon.

File: webapp/src/share.tsx [hunk 1]
```diff
@@ -156,6 +157,28 @@ pxt extract ${url}`;
...
+            const twitterHandle = pxt.appTarget.appTheme.twitterHandle;
+            const orgTwitterHandle = pxt.appTarget.appTheme.orgTwitterHandle;
+            if (twitterHandle && orgTwitterHandle) {
+                twitterText = lf("Check out what I made with {0} and {1}!", twitterHandle, orgTwitterHandle);
+            } else if (twitterHandle) {
+                twitterText = lf("Check out what I made with {0}!", twitterHandle);
+            } else if (orgTwitterHandle) {
+                twitterText = lf("Check out what I made with {0}!", orgTwitterHandle);
+            }
+            fbUrl = `https://www.facebook.com/sharer/sharer.php?u=${url}`;
```
Comment (pelikhan) [inline webapp/src/share.tsx]: Don't you have to encode the URL?

File: webapp/src/share.tsx [hunk 2]
```diff
@@ -156,6 +157,28 @@ pxt extract ${url}`;
...
+            twitterUrl = `https://twitter.com/intent/tweet?url=${url}&text=${encodeURIComponent(twitterText)}`;
+        }
+
+        const popupWindow = (url: string, title: string, width: number, height: number) => {
```
Comment (pelikhan) [inline webapp/src/share.tsx]: Move to sui.tsx

**Comment by pelikhan** [comment]
Should be able to specify default hashtag like "#microbit"

**Comment by microsoftsam** [comment]
PR ready.

**Comment by microsoftsam** [comment]
Using one field: "socialOptions" to indicate we want to see the social icons in the share dialog: 
```
    interface SocialOptions {
        twitterHandle?: string;
        orgTwitterHandle?: string;
        hashtags?: string;
        related?: string;
    }
```

**Comment by pelikhan** [comment]
Analytics, do people even click those buttons?

**Comment by abchatra** [comment]
Works on all browsers?

**Comment by microsoftsam** [comment]
Not doing anything browser specific, so should do, but let's do a sanity check on Windows browsers.

**Comment by microsoftsam** [comment]
Tested on Edge. everything looks fine.

**Pull request merged**